trainer/derrian_backend/custom_scheduler/LoraEasyCustomOptimizer/abmog.py

```
@@ -0,0 +1,527 @@
+                        history_gpu = [g for g in history]
+
+                        # Predictor (Adams-Bashforth)
+                        ab_c = self.ab_coeffs[current_k]
+                        p_pred = torch.zeros_like(state["p_history"][0])
+                        for i in range(current_k):
+                            p_pred.add_(history_gpu[i], alpha=ab_c[i])
+
+                        # Corrector (Adams-Moulton)
+                        am_c = self.am_coeffs[current_k]
+                        # Use predicted grad as proxy for g_{n+1}
+                        corrector_hist = [p_pred] + history_gpu[:-1]
```

Ruff (0.15.13)

[warning] Consider `[p_pred, *history_gpu[:-1]]` instead of concatenation

Replace with `[p_pred, *history_gpu[:-1]]`

(RUF005)